internal/services/hybridcompute/arc_machine_extension_resource.go

```diff
@@ -0,0 +1,549 @@
+				validation.StringDoesNotContainAny("/"),
+			),
+		},
+
+		"arc_machine_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: machines.ValidateMachineID,
+		},
+
+		"automatic_upgrade_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+
+		"force_update_tag": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"location": commonschema.Location(),
+
+		"protected_settings": {
+			Type:             pluginsdk.TypeString,
+			Optional:         true,
+			Sensitive:        true,
+			ValidateFunc:     validation.StringIsJSON,
+			DiffSuppressFunc: pluginsdk.SuppressJsonDiff,
+		},
+
+		"publisher": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"settings": {
+			Type:             pluginsdk.TypeString,
+			Optional:         true,
+			ValidateFunc:     validation.StringIsJSON,
+			DiffSuppressFunc: pluginsdk.SuppressJsonDiff,
+		},
+
+		"type": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"type_handler_version": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+				// suppress any diff if automatic_upgrade_enabled is true
+				if value, ok := d.GetOk("automatic_upgrade_enabled"); ok && value.(bool) {
+					return true
+				}
+				// e.g. 1.24 -> 1.24.1 will be considered as no change
+				if len(oldValue) > 0 && len(newValue) > 0 && strings.HasPrefix(oldValue, newValue) {
+					return true
+				}
+				return false
+			},
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r ArcMachineExtensionResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"instance_view": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"status": {
+						Type:     pluginsdk.TypeList,
+						Computed: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"code": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+
+								"display_status": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+
+								"level": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+
+								"message": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+
+								"time": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
```

Comment:
Is it possible to have multiple statuses? These properties can probably be flattened and moved up a level